File: microservices/user-service/src/main/java/com/saladin/dto/Event.java
package com.saladin.dto;

import lombok.Data;
import lombok.Builder;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Event {
    private Long id;
    private String title;
    private String description;
    private LocalDateTime date;
    private String place;
    private Double price;
    private Long organizerId;
    private String imageUrl;
    private Integer nbPlaces;
    private Integer nbLikes;
    private List<String> domaines;
    private String status;
    private boolean archived;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: frontend/src/app/models/event.ts
export type EventStatus = 'PUBLISHED' | 'DRAFT' | 'CANCELLED' | 'COMPLETED';

export interface Event {
  id: number;
  title: string;
  description: string;
  date: string;
  place: string;
  price: number;
  organizerId: number;
  imageUrl: string;
  nbPlaces: number;
  nbLikes: number;
  domaines: string[];
  status: EventStatus;
  archived: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface EventRequest {
  title: string;
  description: string;
  date: string;
  place: string;
  price: number;
  organizerId: number;
  imageUrl?: string;
  nbPlaces: number;
  nbLikes?: number;
  domaines?: string[];
  status?: EventStatus;
}
